cmd/controller-manager/app/options/options.go

```diff
@@ -195,63 +195,93 @@ func NewGenericControllerManagerOptions(componentConfig componentconfig.KubeCont
-		ConcurrentRCSyncs:                               5,
-		ConcurrentRSSyncs:                               5,
-		ConcurrentDaemonSetSyncs:                        2,
-		ConcurrentJobSyncs:                              5,
-		ConcurrentResourceQuotaSyncs:                    5,
-		ConcurrentDeploymentSyncs:                       5,
-		ConcurrentNamespaceSyncs:                        10,
-		ConcurrentSATokenSyncs:                          5,
-		RouteReconciliationPeriod:                       metav1.Duration{Duration: 10 * time.Second},
-		ResourceQuotaSyncPeriod:                         metav1.Duration{Duration: 5 * time.Minute},
-		NamespaceSyncPeriod:                             metav1.Duration{Duration: 5 * time.Minute},
-		PVClaimBinderSyncPeriod:                         metav1.Duration{Duration: 15 * time.Second},
-		HorizontalPodAutoscalerSyncPeriod:               metav1.Duration{Duration: 30 * time.Second},
-		HorizontalPodAutoscalerUpscaleForbiddenWindow:   metav1.Duration{Duration: 3 * time.Minute},
-		HorizontalPodAutoscalerDownscaleForbiddenWindow: metav1.Duration{Duration: 5 * time.Minute},
-		HorizontalPodAutoscalerTolerance:                0.1,
-		DeploymentControllerSyncPeriod:                  metav1.Duration{Duration: 30 * time.Second},
-		MinResyncPeriod:                                 metav1.Duration{Duration: 12 * time.Hour},
-		RegisterRetryCount:                              10,
-		PodEvictionTimeout:                              metav1.Duration{Duration: 5 * time.Minute},
-		NodeMonitorGracePeriod:                          metav1.Duration{Duration: 40 * time.Second},
-		NodeStartupGracePeriod:                          metav1.Duration{Duration: 60 * time.Second},
-		NodeMonitorPeriod:                               metav1.Duration{Duration: 5 * time.Second},
-		ClusterName:                                     "kubernetes",
-		NodeCIDRMaskSize:                                24,
-		ConfigureCloudRoutes:                            true,
-		TerminatedPodGCThreshold:                        12500,
-		VolumeConfiguration: componentconfig.VolumeConfiguration{
-			EnableHostPathProvisioning: false,
-			EnableDynamicProvisioning:  true,
-			PersistentVolumeRecyclerConfiguration: componentconfig.PersistentVolumeRecyclerConfiguration{
-				MaximumRetry:             3,
-				MinimumTimeoutNFS:        300,
-				IncrementTimeoutNFS:      30,
-				MinimumTimeoutHostPath:   60,
-				IncrementTimeoutHostPath: 30,
+		GenericComponent: componentconfig.GenericComponentConfiguration{
+			Port:                      insecurePort,
+			Address:                   "0.0.0.0",
+			RouteReconciliationPeriod: metav1.Duration{Duration: 10 * time.Second},
+			MinResyncPeriod:           metav1.Duration{Duration: 12 * time.Hour},
+			NodeMonitorPeriod:         metav1.Duration{Duration: 5 * time.Second},
+			ClusterName:               "kubernetes",
+			ConfigureCloudRoutes:      true,
+			ContentType:               "application/vnd.kubernetes.protobuf",
```

Comment:
where is this used?

Reply:
Struct `GenericComponentConfigOptions`  aims to put  some parameter both kube-controller-manager and cloud -controller-manager used. For example, `NodeMonitorPeriod` used in cloud-node-controller and nodelifeController, i think is difficulty to order it to specially  controller,  so just put it in `GenericComponentConfigOptions`. WDYT?

Reply:
Have you checked all of these to be used in both controller managers?

I would prefer to have clearly generic values only here in the struct. E.g. MinResyncPeriod, ContentType, KubeAPIQPS, KubeAPIBurst, ControllerStartInterval, LeaderElection look generic. Think of we create another controller-manager with different controllers. Which values will it need as well?

All the others should probably be duplicated in both binaries.

Reply:
emmm, little confused about it. Both controller-manager used it. I think it should better construct more well struct to split some parameter out `GenericComponentConfigOptions` to make `GenericComponentConfigOptions` more Generic.

Reply:
There is a difference between "generic" and "shared" options. The first are semantically generic options. The later are those that are shared because we would duplicate implementation otherwise.

Imagine you have a 3rdparty-controller-manager, managing external 3rdparty resources, e.g. a PaaS or Istio. Which options would they expect to have as well ("generic")? Which will be kube specific ("shared" between kube and cloud ctrl-manager) ?

Reply:
thank for explanations, split it now